ios/Classes/CountlyFlutterPlugin.m

```diff
@@ -359,7 +359,46 @@ - (void)handleMethodCall:(FlutterMethodCall*)call result:(FlutterResult)result {
         result(@"setLocation!");
         });
 
-    }else if ([@"enableCrashReporting" isEqualToString:call.method]) {
+    }else if ([@"setUserLocation" isEqualToString:call.method]) {
+         dispatch_async(dispatch_get_main_queue(), ^ {
+            NSDictionary* location = [command objectAtIndex:0];
+             NSString* gpsCoordinates = location[@"gpsCoordinates"];
+            CLLocationCoordinate2D locationCoordinate = kCLLocationCoordinate2DInvalid;
+            if(gpsCoordinates){
+                if([gpsCoordinates containsString:@","]) {
+                    @try{
+                        NSArray *locationArray = [gpsCoordinates componentsSeparatedByString:@","];
+                        if(locationArray.count > 2) {
+                            COUNTLY_FLUTTER_LOG(@"Invalid location Coordinates: '%@', it should contains only two comma seperated values", gpsCoordinates);
+                        }
+                        NSString* latitudeString = [locationArray objectAtIndex:0];
+                        NSString* longitudeString = [locationArray objectAtIndex:1];
+
+                        double latitudeDouble = [latitudeString doubleValue];
+                        double longitudeDouble = [longitudeString doubleValue];
+                        if(latitudeDouble == 0 || longitudeDouble == 0) {
+                            COUNTLY_FLUTTER_LOG(@"Invalid location Coordinates: '%@', it should contains two comma seperated double values", gpsCoordinates);
```

Comment:
this warning needs to be reworked

Reply:
warn about the 0 in the coordinates

Reply:
"One of the values parsed to a 0, double check that given coordinates are correct:[$coordinates]."